### PR TITLE
Adds support for use case templates

### PR DESCRIPTION
**Pull request by hiimbex**
In support of https://github.com/probot/probot/issues/674, this PR aims to accomplish the goals of how the create-probot-app script will access the use case templates. It does so by referencing them locally, and will include them in the npm module.

Right now this PR also includes the code beginnings for several of the use case templates. I'll follow up with separate PRs to update those and add tests for all of them.

### CODE REVIEW

File: templates/basic-ts/test/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    app.app = () => 'test'
+  })
+
+  test('creates a passing check', async () => {
+    // Test that we correctly return a test token
+    nock('https://api.github.com')
+      .post('/app/installations/2/access_tokens')
+      .reply(200, { token: 'test' })
+
+    // Test that a commented is posted
```
Comment (TimonVS) [inline templates/basic-ts/test/index.test.ts]:
```suggestion
    // Test that a comment is posted
```

File: templates/basic-ts/test/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    nock('https://api.github.com')
+      .post('/repos/hiimbex/testing-things/issues/1/comments', (body: any) => {
+        expect(body).toMatchObject(issueCreatedBody)
+        return true
+      })
+      .reply(200)
+
+    // Recieve a webhook event
```
Comment (TimonVS) [inline templates/basic-ts/test/index.test.ts]:
```suggestion
    // Receive a webhook event
```

File: README.md [hunk 3]
```diff
@@ -1,26 +1,46 @@
...
+This template is a TypeScript app that will post a comment every time an issue is opened.
 
 ```sh
-yarn create probot-app [--typescript] my-first-app
+npx create-probot-app my-first-app -t checks-js
 ```
 
-See the [Probot docs](https://probot.github.io/docs/development/) to get started.
\ No newline at end of file
+This template is a JavaScript app that will create a new [check](https://developer.github.com/v3/checks/) every time a check a push happens.
```
Comment (TimonVS) [inline README.md]:
```suggestion
This template is a JavaScript app that will create a new [check](https://developer.github.com/v3/checks/) every time a push happens.
```

File: templates/basic-js/test/index.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    app.app = () => 'test'
+  })
+
+  test('creates a passing check', async () => {
+    // Test that we correctly return a test token
+    nock('https://api.github.com')
+      .post('/app/installations/2/access_tokens')
+      .reply(200, { token: 'test' })
+
+    // Test that a commented is posted
```
Comment (TimonVS) [inline templates/basic-js/test/index.test.js]:
```suggestion
    // Test that a comment is posted
```

File: templates/basic-js/test/index.test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    nock('https://api.github.com')
+      .post('/repos/hiimbex/testing-things/issues/1/comments', (body) => {
+        expect(body).toMatchObject(issueCreatedBody)
+        return true
+      })
+      .reply(200)
+
+    // Recieve a webhook event
```
Comment (TimonVS) [inline templates/basic-js/test/index.test.js]:
```suggestion
    // Receive a webhook event
```

File: templates/checks-js/test/index.test.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    nock('https://api.github.com')
+      .post('/repos/hiimbex/testing-things/check-runs', (body) => {
+        body.completed_at = '2018-10-05T17:35:53.683Z'
+        expect(body).toMatchObject(checkRunSuccess)
+        return true
+      })
+      .reply(200)
+
+    // Recieve a webhook event
```
Comment (TimonVS) [inline templates/checks-js/test/index.test.js]:
```suggestion
    // Receive a webhook event
```

File: templates/basic-ts/test/index.test.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  beforeEach(() => {
+    probot = new Probot({ id: 123, cert: 'test' })
+    // Load our app into probot
+    const app = probot.load(myProbotApp)
+
+    // just return a test token
+    app.app = () => 'test'
+  })
+
+  test('creates a passing check', async () => {
```
Comment (TimonVS) [inline templates/basic-ts/test/index.test.ts]:
```suggestion
  test('creates a comment when an issue is opened', async () => {
```

File: templates/basic-js/test/index.test.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  beforeEach(() => {
+    probot = new Probot({})
+    // Load our app into probot
+    const app = probot.load(myProbotApp)
+
+    // just return a test token
+    app.app = () => 'test'
+  })
+
+  test('creates a passing check', async () => {
```
Comment (TimonVS) [inline templates/basic-js/test/index.test.js]:
```suggestion
  test('creates a comment when an issue is opened', async () => {
```

File: templates/deploy-js/test/index.test.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+    // Test that a deployment status is created
+    nock('https://api.github.com')
+      .post('/repos/hiimbex/testing-things/deployments/123/statuses', (body) => {
+        expect(body).toMatchObject(deploymentStatus)
+        return true
+      })
+      .reply(200)
+
+    // Recieve a webhook event
```
Comment (TimonVS) [inline templates/deploy-js/test/index.test.js]:
```suggestion
    // Receive a webhook event
```

**Comment by hiimbex** [comment]
Thanks @TimonVS!! Spelling is not my strong suit 😅

**Pull request merged**